scripts/oneoff/upload_draft_service_pdfs.py

```diff
@@ -0,0 +1,267 @@
+    draft_json = {
+        DOC_TYPES[doc_type]: full_document_url
+    }
+    if dry_run:
+        # Log something
+        print(f"[Dry-run] API: Would update draft ID {draft_id} with some JSON:")
+        print(draft_json)
+        return True
+
+    try:
+        api_client.update_draft_service(
+            draft_id,
+            draft_json,
+            'one off PDF upload script'
+        )
+        print(f"Draft ID {draft_id} updated.")
+        draft = api_client.get_draft_service(draft_id)
+        if draft['validationErrors']:
+            print(f"Validation errors:")
+            print(draft['validationErrors'])
+            return False
+        else:
+            print("Ready to be marked as complete.")
+            return True
+    except Exception as exc:
+        print(str(exc))
+    return False
+
+
+def output_results(unidentifiable_files, successful_draft_ids, failed_draft_ids):
+    # Output successful / failed draft IDs
+    if successful_draft_ids:
+        with open('successful-uploads-draft-ids.txt', 'w') as f:
+            for d in successful_draft_ids:
+                f.write(str(d) + '\n')
+
+    if failed_draft_ids:
+        with open('failed-uploads-draft-ids.txt', 'w') as f:
+            for d in failed_draft_ids:
+                f.write(str(d) + '\n')
+
+    if unidentifiable_files:
+        with open('unidentifiable-draft-service-files.txt', 'w') as f:
+            for d in unidentifiable_files:
+                f.write(str(d) + '\n')
+
+
+def upload_draft_service_pdfs_from_folder(
+    bucket, bucket_category, assets_path, local_directory, api_client, framework_slug, dry_run
+):
+    failed_draft_ids = []
+    successful_draft_ids = []
+    unidentifiable_files = []
+
+    # Parse files in folder (TODO: support ODF files)
+    for path in get_all_files_of_type(local_directory, 'pdf'):
+        # Check file is open document format or PDF
+        file_name = os.path.basename(path)
+        print(f"File name: {file_name}")
+        _, file_extension = os.path.splitext(file_name)
+        if file_extension not in [".pdf", ".odt", ".ods", ".odp"]:
+            print(f'Invalid file type {file_extension}, skipping')
+            unidentifiable_files.append(file_name)
+            continue
+
+        # Get IDs etc
+        supplier_id, doc_type, draft_id = get_info_from_filename(file_name, api_client, framework_slug)
+        if supplier_id is None:
+            print("Unable to determine Supplier ID")
+            unidentifiable_files.append(file_name)
+            continue
+        if doc_type is None:
+            print("Unable to determine doctype")
+            unidentifiable_files.append(file_name)
+            continue
+        if draft_id is None:
+            print("Unable to determine Draft ID")
+            unidentifiable_files.append(file_name)
+            continue
+        print(f"Found Supplier ID {supplier_id}, file type {doc_type} and Draft ID {draft_id}")
```

Comment:
Sometimes filenames have erroneous whitespace which would cause the supplier ID to be invalid.

```suggestion
        # Strip erroneous whitespace from supplier IDs
        supplier_id = supplier_id.strip()
        print(f"Found Supplier ID {supplier_id}, file type {doc_type} and Draft ID {draft_id}")
```

Reply:
This is now handled in the `get_info_from_filename` function where we cast `supplier_id` to an int, hopefully that'll get rid of any pesky whitespace.